### PR TITLE
Finalise Action for Facility Catchment + Validations

### PR DESCRIPTION
**Pull request by rachana-egov**
## Release Notes

- **New Features**
  - Enhanced error handling and user feedback in multiple components, including `WorkflowCommentPopUp`, `AccessibilityPopUp`, `EditVillagePopulationPopUp`, and `SecurityPopUp`, with dynamic toast notifications for better error reporting.
  - Introduced a new route for handling success responses related to village operations in the application.

- **Bug Fixes**
  - Improved robustness in the `PopInbox` component by refining conditions for API calls to prevent potential runtime errors.

- **Improvements**
  - Added role verification and action management in the `FacilityCatchmentMapping` component, enhancing user interaction based on permissions.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js [hunk 1]
```diff
@@ -1,22 +1,143 @@
...
+  const {
+    isLoading: isPlanEmpSearchLoading,
+    data: planEmployee,
+    error: planEmployeeError,
+    refetch: refetchPlanEmployee,
+  } = Digit.Hooks.microplanv1.usePlanSearchEmployeeWithTagging({
+    tenantId: tenantId,
+    body: {
+      PlanEmployeeAssignmentSearchCriteria: {
+        tenantId: "mz",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js:32]: update hardcoded tenantId @rachana-egov

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js [hunk 2]
```diff
@@ -1,22 +1,143 @@
...
+
+  const { isLoading: isLoadingPlanObject, data: planObject, error: errorPlan, refetch: refetchPlan } = Digit.Hooks.microplanv1.useSearchPlanConfig(
+    {
+      PlanConfigurationSearchCriteria: {
+        tenantId,
+        id: url?.microplanId,
+      },
+    },
+    {
+      enabled: isRootApprover && data?.TotalCount < 1,
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js:71]: shouldn't this be data?.TotalCount===0 @rachana-egov

Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js (hunk 1 above)
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js [hunk 3]
```diff
@@ -1,22 +1,143 @@
...
+  return (
+    <React.Fragment>
+      <Header>{t("MICROPLAN_ASSIGN_CATCHMENT_VILLAGES")}</Header>
+      <div className="inbox-search-wrapper">
+        <InboxSearchComposer
+          configs={config}
+        ></InboxSearchComposer>
+      </div>
+
+      {isRootApprover && data?.TotalCount < 1 &&
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/FacilityCatchmentMapping.js]: done

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**